mypy/util.py

```diff
@@ -401,10 +492,18 @@ def colorize(self, error: str) -> str:
         elif ': note:' in error:
             loc, msg = error.split('note:', maxsplit=1)
             return loc + self.style('note:', 'blue') + self.underline_link(msg)
+        elif self.show_source_code and error.startswith(' ' * DEFAULT_SOURCE_OFFSET):
```

Comment:
Detecting source code highlights through an indent prefix can be surprising. I wonder if there's a more general way to do this. If there's no alternative simple approach, this should be documented carefully somewhere.